src/views/ActiveSpeakerView.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from "react";
+import { selectPeers, useHMSStore } from "@100mslive/hms-video-react";
+import { GridCenterView, GridSidePaneView } from "./components/gridView";
+import {
+  selectDominantSpeaker,
+  selectLocalPeer,
+} from "@100mslive/hms-video-store";
+
+export const ActiveSpeakerView = ({
+  isChatOpen,
+  toggleChat,
+  isParticipantListOpen,
+}) => {
+  const peers = useHMSStore(selectPeers);
+  const localPeer = useHMSStore(selectLocalPeer);
+  let [activeSpeaker, setActiveSpeaker] = React.useState(localPeer);
+  let dominantSpeaker = useHMSStore(selectDominantSpeaker);
+  let showSidePane = activeSpeaker && peers.length > 1;
+
+  /** here we are using peer filter function to change the activeSpeaker and sidebarPeers,
+   * on first mount activeSpeaker points to the localPeer and on each update it points
+   * to the dominantSpeaker
+   */
+  const peerFilter = dominantSpeaker => {
+    if (dominantSpeaker) {
```

Comment:
shouldn't we also set localpeer in else if there is no dominant speaker?

Reply:
Initially the localPeer is the active speaker. If there is any dominant speaker, then the activeSpeaker will be set to this dominantSpeaker but in case there is none we have to show the previous dominantSpeaker. So else if condition won't be necessary because we don't have to set it back to localPeer.

Reply:
@raviteja83 dominant speaker might take pause between speaking, showing local peer will lead to flicker and bad ux